jobs/schedulers.go

```diff
@@ -114,6 +115,14 @@ func (schedulers *Schedulers) Start() *Schedulers {
 							schedulers.setNextRunTime(newCfg, idx, now, false)
 						}
 					}
+				case isLeader := <-schedulers.clusterLeaderChanged:
+					for idx := range schedulers.schedulers {
+						if !isLeader {
+							schedulers.nextRunTimes[idx] = nil
+						} else {
+							schedulers.setNextRunTime(schedulers.jobs.Config(), idx, now, false)
```

Comment:
Does this reset the runtime whenever the cluster leader changes, or does it take into account something in the database to compute the last run time? Wondering if a rash of cluster leader changes (or a bad connection to a node) would trigger repeated runs.

Reply:
The `setNextRunTime()` method on the scheduler figures out when the next run time should be, rather than actually running anything. In the cases we have at the moment, they run at a particular time of day, so instability wouldn't result in any extra jobs being scheduled.

Reply:
And some more detail: determining when to schedule the next job *is* stateful, so it shouldn't result in any issues with other hypothetical future schedulers too.